### DISCUSSION TITLE
Error Call to a member function getKey() on nul when overloading or extending DeleteAction for header in EditPage

### DISCUSSION THREAD

**Discussion by bvicini** [filamentphp/filament#13590] [Help] [answered]
### Package

Actions

### Package Version

v3.2.93

### How can we help you?

I'm trying to overload the default DeleteAction in order to catch an integrity violation exception and handle it correctly,  I use the following code in getHeaderActions() of the resource EditRecord  file

\Filament\Actions\DeleteAction::make()
            ->action(function($record) {
                $record->delete() ;
            }),

it is a minimal code, but when I execute this, the error "Call to a member function getKey() on null" arises.
The error occurs even if I create a custom Action that extends DeleteAction or even if I copy the entire file DeleteAction from vendor folder. If I modify the vendor file itself the error does not show up.

**Comment by leandrocfe** [comment, marked as answer]
If you want to extend this action, you should do this

```php
\Filament\Actions\DeleteAction::make()
    ->action(function (\Filament\Actions\DeleteAction $action) {

        $result = $action->process(static fn (Model $record) => $record->delete());

        if (! $result) {
            $action->failure();

            return;
        }

        $action->success();
    }),
```

**Reply by bvicini** [reply]
Thank you very much, it works. Can you suggest me where to find these indications in the docs? I had to search in internet and copied the code from there :(. Ty again
